core/chains/evm/txmgr/orm.go

```diff
@@ -540,3 +575,182 @@ WHERE eth_tx_attempts.state = 'in_progress' AND eth_txes.from_address = $1 AND e
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		err = tx.Select(&etxs, `
+SELECT DISTINCT eth_txes.* FROM eth_txes
+INNER JOIN eth_tx_attempts ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_tx_attempts.state = 'broadcast'
+INNER JOIN eth_receipts ON eth_receipts.tx_hash = eth_tx_attempts.hash
+WHERE eth_txes.state IN ('confirmed', 'confirmed_missing_receipt') AND block_number BETWEEN $1 AND $2 AND evm_chain_id = $3
+ORDER BY nonce ASC
+`, lowBlockNumber, highBlockNumber, chainID.String())
+		if err != nil {
+			return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_txes")
+		}
+		if err = loadEthTxesAttempts(tx, etxs); err != nil {
+			return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_tx_attempts")
+		}
+		err = loadEthTxesAttemptsReceipts(tx, etxs)
+		return errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed to load eth_receipts")
+	}, pg.OptReadOnlyTx())
+	return etxs, errors.Wrap(err, "FindTransactionsConfirmedInBlockRange failed")
+}
+
+func (o *orm) saveAttemptWithNewState(q pg.Queryer, timeout time.Duration, attempt EthTxAttempt, broadcastAt time.Time) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	return pg.SqlxTransaction(ctx, q, o.logger, func(tx pg.Queryer) error {
+		// In case of null broadcast_at (shouldn't happen) we don't want to
+		// update anyway because it indicates a state where broadcast_at makes
+		// no sense e.g. fatal_error
+		if _, err := tx.Exec(`UPDATE eth_txes SET broadcast_at = $1 WHERE id = $2 AND broadcast_at < $1`, broadcastAt, attempt.EthTxID); err != nil {
+			return errors.Wrap(err, "saveAttemptWithNewState failed to update eth_txes")
+		}
+		_, err := tx.Exec(`UPDATE eth_tx_attempts SET state=$1 WHERE id=$2`, attempt.State, attempt.ID)
+		return errors.Wrap(err, "saveAttemptWithNewState failed to update eth_tx_attempts")
+	})
+}
+
+func (o *orm) SaveInsufficientEthAttempt(timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error {
+	if !(attempt.State == EthTxAttemptInProgress || attempt.State == EthTxAttemptInsufficientEth) {
+		return errors.New("expected state to be either in_progress or insufficient_eth")
+	}
+	attempt.State = EthTxAttemptInsufficientEth
+	return errors.Wrap(o.saveAttemptWithNewState(o.q, timeout, *attempt, broadcastAt), "saveInsufficientEthAttempt failed")
+}
+
+func (o *orm) saveSentAttempt(q pg.Queryer, timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error {
+	if attempt.State != EthTxAttemptInProgress {
+		return errors.New("expected state to be in_progress")
+	}
+	attempt.State = EthTxAttemptBroadcast
+	return errors.Wrap(o.saveAttemptWithNewState(q, timeout, *attempt, broadcastAt), "saveSentAttempt failed")
+}
+
+func (o *orm) SaveSentAttempt(timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error {
```

Comment:
It's unclear to me why this ever took a timeout. Should take a context?

Reply:
I suppose its because it's only ever used to create a new context with a timeout. It could take a context as well which would make it more general-purpose

Reply:
Yeah this looks like legacy/oversight